ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py

```diff
@@ -0,0 +1,970 @@
+
+        if self.utility.node_type == "MASTER":
+
+            cluster_status = self.utility.execute_bash_command("rabbitmqctl cluster_status")
+
+            if "Error" not in cluster_status[1]:
+
+                # this can happen
+                if len(cluster_status) <= 3:
+                    partition_status = cluster_status[2]
+                else:
+                    partition_status = cluster_status[3]
+
+                # check parition status
+                if '@' in partition_status:
+                    self.LOGGER.logger("Seems like the RabbitMQ cluster has 'partition' problems, please check this...",
+                                       self.module, 0, 'process_rabbitmq', False)
+                else:
+                    self.LOGGER.logger("RabbitMQ does not seem to have 'partition' problems :D", self.module, 1,
+                                       'process_rabbitmq', False)
+            else:
+                self.LOGGER.logger("Seems like the RabbitMQ cluster has errors, maybe it is offline?", self.module, 0,
+                                   'process_rabbitmq', False)
+
+        else:
+            self.LOGGER.logger("RabbitMQ is not running/active on this server!", self.module, 5,
+                               'process_rabbitmq', False)
+
+        #
+        # Checking consistency of volumedriver vs. ovsdb and backwards
+        #
+
+        for vp in VPoolList.get_vpools():
+
+            self.LOGGER.logger("Checking consistency of volumedriver vs. ovsdb for vPool '{0}': ".format(vp.name),
+                               self.module, 3, 'checkDiscrepanciesVoldrvOvsdb', False)
+
+            # list of vdisks that are in model but are not in volumedriver
+            missinginvolumedriver = []
+
+            # list of volumes that are in volumedriver but are not in model
+            missinginmodel = []
+
+            # fetch configfile of vpool for the volumedriver
+            config_file = self.utility.get_config_file_path(vp.name, self.machine_id, 1, vp.guid)
+            voldrv_client = src.LocalStorageRouterClient(config_file)
+
+            # collect data from volumedriver
+            voldrv_volume_list = voldrv_client.list_volumes()
+
+            # collect data from model
+            model_vdisk_list = vp.vdisks
+
+            # crossreference model vs. volumedriver
+            for vdisk in model_vdisk_list:
+                if not vdisk.volume_id in voldrv_volume_list:
+                    missinginvolumedriver.append(vdisk.volume_id)
+
+            # crossreference volumedriver vs. model
+            # (This can be a performance bottleneck on heavy env. due to nested for loops)
+            for volume in voldrv_volume_list:
+                for vdisk in model_vdisk_list:
+                    if str(volume) != vdisk.volume_id and len(model_vdisk_list) == (model_vdisk_list.index(vdisk)+1):
+                        missinginmodel.append(volume)
```

Comment:
Wrt to finding out which vDisks that are not in the model but known to the StorageDriver and vice versa, sth like this code might be more efficient:

```
model_volumes = set(vdisk.volume_id for vdisk in vp.disks)
storagedriver_volumes = set(str(volume) for volume in voldrv_client.list_volumes())
missinginvolumedriver = list(model_volumes - storagedriver_volumes)
missinginmodel = list(storagedriver_volumes - model_volumes)
```